analysis/export_metrics_to_xlsx.py

```diff
@@ -0,0 +1,131 @@
+import copy
+
+import classes
+import decision
+import decision.value_functions
+import clustering.scripts
+import errno
+import pandas as pd
+import os
+from openpyxl import load_workbook
+from personal import *
+
+
+def metrics_to_xlsx(instances: [classes.World]):
+    """
+    Method to export metrics for a list of evaluated instances to Excel
+    :param instances: list of instances to be exported
+    """
+    # assuming that all instances is tested on the same parameter
+    parameter_name = instances[0].metrics.testing_parameter_name
+    column_tuples = []
+    metrics_data = []
+
+    # loop through all instances and record their metrics
+    for instance in instances:
+        metrics_data, column_tuples = add_metric_column(
+            instance, column_tuples=column_tuples, metrics_data=metrics_data
+        )
+
+    # creating the directory to save the file if it doesn't exist
+    try:
+        os.makedirs(PATH_COMPUTATIONAL_STUDY)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            raise
+
+    file_name = f"{PATH_COMPUTATIONAL_STUDY}/{parameter_name.title()}.xlsx"
+
+    # if the file isn't created -> create a new .xlsx file
+    if not os.path.isfile(file_name):
+        pd.DataFrame().to_excel(file_name)
+    book = load_workbook(file_name)
+    writer = pd.ExcelWriter(file_name, engine="openpyxl")
+    writer.book = book
+    writer.sheets = dict((ws.title, ws) for ws in book.worksheets)
+
+    sheets = [ws.title.split("_")[0] for ws in book.worksheets]
+
+    try:
+        # have to replace : since its illegal that a sheet name contains the character
+        instance_created_at = instances[0].created_at.replace(":", ".")
```

Comment:
You are resuing `instance[0]`, maybe create a separate variable for this one?

Reply:
Thought about it. Here I assume that all instances are tested from the same world instance and aims to evaluate the same parameter (or none, e.g DoNothing, ...). Can take this as an input argument, but thought it was cleaner to only take a list of instances in as input, and handle everything else inside the method